spec/syntax.md

```diff
@@ -227,19 +227,34 @@ Any such future keyword must start with `.`,
 followed by two or more lower-case ASCII characters.
 
 The rest of the statement supports
-a similarly wide range of content as _reserved annotations_,
-but it MUST end with one or more _expressions_.
+a similarly wide range of content as _reserved annotations_ (with the exception
+of U+002E FULL STOP `.`, which is only allowed following characters that are
+permissible in _name_), but it MUST end with one or more _expressions_.
 
 ```abnf
-reserved-statement = reserved-keyword [s reserved-body] 1*([s] expression)
+reserved-statement = reserved-keyword [s reserved-statement-body] 1*([s] expression)
 reserved-keyword   = "." name
+
+reserved-statement-body      = reserved-statement-body-part *([s] reserved-statement-body-part)
+reserved-statement-body-part = (name-char-no-dot 1*".") / content-char
+                             / escaped-char / quoted-literal
 ```
 
 > [!NOTE]
 > The `reserved-keyword` ABNF rule is a simplification,
 > as it MUST NOT be considered to match any of the existing keywords
 > `.input`, `.local`, or `.match`.
 
+> [!NOTE]
+> Only dots that are preceded by a valid name character are allowed in a
+> `reserved-statement-body`. This enables better parser error recovery for tools in the
+> case of incomplete reserved statements, like in the following example:
+> ```mf2
```

Comment:
Semantic line breaks, and `mf2` isn't a recognised linguist syntax identifier at least yet. If we're to add that, let's do so systematically everywhere.

```suggestion
> Only dots that are preceded by a valid name character are allowed in a `reserved-statement-body`.
> This enables better parser error recovery for tools
> in the case of incomplete reserved statements,
> like in the following example:
> ```
```